src/tools/clippy/tests/ui/significant_drop_in_scrutinee.stderr

```diff
@@ -1,113 +1,135 @@
-error: The lifetimes of temporaries in a match scrutinee last to the end of the match expressions. One or more of the temporaries in this match scrutinee have a drop with a side-effect (such as unlocking a mutex). Consider calling the scrutinee outside of the match expression and using the result instead.
+error: temporary with significant drop in match scrutinee
   --> $DIR/significant_drop_in_scrutinee.rs:46:11
    |
 LL |     match mutex.lock().unwrap().foo() { // Should trigger lint
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
It would be nice to restrict the span of the lint to the part of the expression that creates the temporary. So in this case

```suggestion
LL |     match mutex.lock().unwrap().foo() { // Should trigger lint
   |           ^^^^^^^^^^^^
```

Reply:
That's a good idea! I would like your thoughts on how this is displayed -- I have it so the error message indicates the span up to and including the point where the temporary with significant drop is. However, the suggestion includes the next item in the chain (so that the temporary will get dropped). Would it be better to just point to the span that's getting moved? I'm concerned it will be confusing without more explanation that might be appropriate for this.